data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateRoot.java

```diff
@@ -75,16 +106,37 @@ public GetStateRoot(final DataProvider dataProvider, final JsonProvider jsonProv
       })
   @Override
   public void handle(@NotNull final Context ctx) throws Exception {
-    final Map<String, String> pathParamMap = ctx.pathParamMap();
-    final SafeFuture<Optional<ObjectAndMetaData<Root>>> future =
-        chainDataProvider.getStateRoot(pathParamMap.get(PARAM_STATE_ID));
-    handleOptionalResult(ctx, future, this::handleResult, SC_NOT_FOUND);
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final SafeFuture<Optional<ObjectAndMetaData<Bytes32>>> future =
+        chainDataProvider.getStateRootBytes32(request.getPathParameter(PARAMETER_STATE_ID));
+    request.handleOptionalResult(future, this::handleResult, SC_NOT_FOUND);
```

Comment:
I think this is where we need to use `request.respondAsync`
A really good example is in `GetNewBlindedBlock`, and the advantage is you're returning the object and letting the request framework handle setting bodies etc... ultimately we should rarely have to use `JsonUtil.serialize` at this level...